src/formatDate.js

```diff
@@ -8,7 +8,84 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatObject = arrayInObject(fromFormat);
+  const toFormatObject = arrayInObject(toFormat);
+  const arrayOldDate = date.split(fromFormat[3]);
+  const dateObject = arrayDateFromObject(arrayOldDate, fromFormat);
+
+  if (fromFormatObject.yy !== toFormatObject.yy) {
```

Comment:
The `fromFormatObject` and `toFormatObject` variables are objects with properties `yy`, `mm`, and `dd`. The condition `fromFormatObject.yy !== toFormatObject.yy` will always be `true` since the properties are strings and you are checking for inequality. You should compare the lengths of the `yy` strings instead to determine if year conversion is needed.